lib/improver/tests/test_nbhood_basicneighbourhoodprocessing.py

```diff
@@ -163,7 +167,79 @@ def set_up_cube_lat_long(zero_point_indices=((0, 7, 7),), num_time_points=1,
     return cube
 
 
-class Test_operation_radius_to_grid_cells(IrisTest):
+class Test__init__(IrisTest):
+
+    def test_radii_varying_with_lead_time_mismatch(self):
+        """
+        Test that the desired error message is raised, if there is a mismatch
+        between the number of radii and the number of lead times.
+        """
+        cube = set_up_cube(num_time_points=3)
+        iris.util.promote_aux_coord_to_dim_coord(cube, "time")
+        time_points = cube.coord("time").points
+        fp_points = [2, 3, 4]
+        cube = _add_forecast_reference_time_and_forecast_period(
+            cube, time_point=time_points, fp_point=fp_points)
+        radii_in_km = [10, 20, 30]
+        lead_times = [2, 3]
+        msg = "There is a mismatch in the number of radii"
+        with self.assertRaisesRegexp(ValueError, msg):
+            NBHood(radii_in_km, lead_times)
+
+
+class Test__find_required_lead_times(IrisTest):
+
+    """Test determining of the lead times present within the input cube."""
+
+    RADIUS_IN_KM = 6.1
+
+    def test_basic(self):
+        """Test that a list is returned."""
+        cube = _add_forecast_reference_time_and_forecast_period(set_up_cube())
+        plugin = NBHood(self.RADIUS_IN_KM)
+        result = plugin._find_required_lead_times(cube)
+        self.assertIsInstance(result, np.ndarray)
+
+    def test_check_coordinate(self):
+        """
+        Test that the data within the list is as expected, when
+        the input cube has a forecast_period coordinate.
+        """
+        cube = _add_forecast_reference_time_and_forecast_period(set_up_cube())
+        expected_result = cube.coord("forecast_period").points
+        plugin = NBHood(self.RADIUS_IN_KM)
+        result = plugin._find_required_lead_times(cube)
+        self.assertArrayAlmostEqual(result, expected_result)
+
+    def test_check_coordinate_without_forecast_period(self):
+        """
+        Test that the data within the list is as expected, when
+        the input cube has a time coordinate and a forecast_reference_time
+        coordinate.
+        """
+        cube = _add_forecast_reference_time_and_forecast_period(set_up_cube())
+        cube.remove_coord("forecast_period")
+        expected_result = (
+            cube.coord("time").points -
+            cube.coord("forecast_reference_time").points)
+        plugin = NBHood(self.RADIUS_IN_KM)
+        result = plugin._find_required_lead_times(cube)
+        self.assertArrayAlmostEqual(result, expected_result)
```

Comment:
Added unit conversion of lead time into a common lead time unit, and added unit tests.